src/System.Management.Automation/help/CommandHelpProvider.cs

```diff
@@ -511,16 +512,22 @@ private string GetHelpFile(string helpFile, CmdletInfo cmdletInfo)
                     // we have to search only in the application base for a mshsnapin...
                     // if you create an absolute path for helpfile, then MUIFileSearcher
                     // will look only in that path.
-                    helpFileToLoad = Path.Combine(mshSnapInInfo.ApplicationBase, helpFile);
+                    
+                    searchPaths.Add(mshSnapInInfo.ApplicationBase);
+                    searchPaths.Add(HelpUtils.GetUserHomeHelpSearchPath());
                 }
                 else if (cmdletInfo.Module != null && !string.IsNullOrEmpty(cmdletInfo.Module.Path))
-                {
-                    helpFileToLoad = Path.Combine(cmdletInfo.Module.ModuleBase, helpFile);
+                {                    
+                    searchPaths.Add(cmdletInfo.Module.ModuleBase);
+                    searchPaths.Add(HelpUtils.GetModuleBaseForUserHelp(cmdletInfo.Module.ModuleBase, cmdletInfo.Module.Name));
                 }
                 else
                 {
                     searchPaths.Add(GetDefaultShellSearchPath());
                     searchPaths.Add(GetCmdletAssemblyPath(cmdletInfo));
+
+                    var homeModulePath = Path.Combine(HelpUtils.GetUserHomeHelpSearchPath(), cmdletInfo.ModuleName);
```

Comment:
> cmdletInfo.ModuleName

I believe `cmdletInfo.ModuleName` will be null when the code reaches here.
Maybe we should only add `HelpUtils.GetUserHomeHelpSearchPath()`? Just like the `GetDefaultShellSearchPath`,

Reply:
Sorry missed that in the last update. Fixed now.